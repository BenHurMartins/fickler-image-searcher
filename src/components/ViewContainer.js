import React from 'react';
import {SafeAreaView, StyleSheet} from 'react-native';
import {Colors, Mixins} from '../styles';

const ViewContainer = props => {
  return (
    <SafeAreaView style={[styles.container, props.styles]}>
      {props.children}
    </SafeAreaView>
  );
};

export default ViewContainer;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.BACKGROUND_COLOR,
    width: Mixins.DEVICE_WIDTH,
  },
});
