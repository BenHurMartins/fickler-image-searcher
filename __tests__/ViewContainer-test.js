import 'react-native';
import React from 'react';
import {Text} from 'react-native';
import {render, act, fireEvent} from '@testing-library/react-native';
import ViewContainer from '../src/components/ViewContainer';
import renderer from 'react-test-renderer';

const children = <Text>test</Text>;
describe('ViewContainer component', () => {
  it('ViewContainer renders correctly', () => {
    renderer.create(<ViewContainer>{children}</ViewContainer>);
  });
  it('ViewContainer snapshot test', () => {
    const tree = renderer
      .create(<ViewContainer>{children}</ViewContainer>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should show history terms', () => {
    const {getByText, getByTestId} = render(
      <ViewContainer>{children}</ViewContainer>,
    );
    act(() => {
      expect(getByText('test')).not.toBe(null);
    });
  });
});
