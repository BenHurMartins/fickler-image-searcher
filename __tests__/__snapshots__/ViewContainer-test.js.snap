// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewContainer component ViewContainer snapshot test 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Text>
    test
  </Text>
</RCTSafeAreaView>
`;
